File: frontend/dto/baseAlbumWithoutSongsResponse.ts
import { BaseArtistResponseSchema } from "@/dto";
import { z } from "zod";

export const BaseAlbumWithoutSongsResponseSchema = z.object({
    type: z.union([z.literal("album")]),
    provider: z.string(),
    publicId: z.string(),
    url: z.string(),
    name: z.string(),
    artists: z.array(z.lazy(() => BaseArtistResponseSchema)),
    releaseDate: z.string(),
    internalImageUrl: z.string(),
});

export type BaseAlbumWithoutSongsResponse = z.infer<
    typeof BaseAlbumWithoutSongsResponseSchema
>;

File: backend/core/responses/baseAlbumWithoutSongsResponse.py
from typing import Literal, Sequence

from pydantic import BaseModel

from backend.core.responses.baseArtistResponse import BaseArtistResponse


class BaseAlbumWithoutSongsResponse(BaseModel):
    """Base response model for a song's album information."""

    type: Literal["album"] = "album"
    provider: str
    publicId: str
    name: str
    artists: Sequence[BaseArtistResponse]
    releaseDate: str
    internalImageUrl: str
